Build failing on latest gcc

Outputs from configure and build commands are:
[./waf configure -T debug > configure.log 2>&1
](https://github.com/nillerusr/source-engine/files/7249585/configure.log) [./waf build -v > compile.log 2>&1
](https://github.com/nillerusr/source-engine/files/7249586/compile.log)
Error is 
```
../engine/gl_matsysiface.cpp: In function ‘void DebugDrawLightmapAtCrossHair()’:
../engine/gl_matsysiface.cpp:975:31: error: ordered comparison of pointer with integer zero (‘SurfaceHandle_t’ {aka ‘msurface2_t*’} and ‘int’)
  975 |         if( s_CrossHairSurfID <= 0 )
      |             ~~~~~~~~~~~~~~~~~~^~~~
In file included from ../public/tier1/utlsymbol.h:17,
                 from ../engine/render_pch.h:13,
                 from ../engine/gl_matsysiface.cpp:12:
```

--- a/engine/gl_matsysiface.cpp
+++ b/engine/gl_matsysiface.cpp
@@ -972,7 +972,7 @@ void DebugDrawLightmapAtCrossHair()
 	IMaterial *pMaterial;
 	int lightmapPageSize[2];
 
-	if( s_CrossHairSurfID <= 0 )
+	if( s_CrossHairSurfID <= (SurfaceHandle_t)0 )
 	{
 		return;
 	}

--- a/game/client/clientmode_shared.cpp
+++ b/game/client/clientmode_shared.cpp
@@ -1152,7 +1152,7 @@ void ClientModeShared::FireGameEvent( IGameEvent *event )
 			}
 		}
 
-		if ( team == 0 && GetLocalTeam() > 0 )
+		if ( team == 0 && GetLocalTeam() > (C_Team*)0 )
 		{
 			bValidTeam = false;
 		}

--- a/materialsystem/shaderapidx9/shaderdevicedx8.cpp
+++ b/materialsystem/shaderapidx9/shaderdevicedx8.cpp
@@ -619,7 +619,7 @@ bool CShaderDeviceMgrDx8::ComputeCapsFromD3D( HardwareCaps_t *pCaps, int nAdapte
 		if ( pVendorID )
 		{
 			int nVendorID = V_atoi( pVendorID );	// use V_atoi for hex support
-			if ( pVendorID > 0 )
+			if ( pVendorID > (HardwareCaps_t*)0 )
 			{
 				ident.VendorId = nVendorID;
 			}

--- a/serverbrowser/CustomGames.cpp
+++ b/serverbrowser/CustomGames.cpp
@@ -211,7 +211,7 @@ bool CCustomGames::CheckTagFilter( gameserveritem_t &server )
 		V_SplitString( m_szTagFilter, ",", TagList );
 		for ( int i = 0; i < TagList.Count(); i++ )
 		{
-			if ( ( Q_strnistr( server.m_szGameTags, TagList[i], MAX_TAG_CHARACTERS ) > 0 ) == TagsExclude() )
+			if ( ( Q_strnistr( server.m_szGameTags, TagList[i], MAX_TAG_CHARACTERS ) > (const char*)0 ) == TagsExclude() )
 			{
 				bRetVal = false;
 				break;

